url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/entities/people/Activity.java

```diff
@@ -0,0 +1,841 @@
+						isShiftable = shiftable.get(WORKING);
+						isExclusive = config.get(WORKING);
+						vector = appliances.get(WORKING);
+					}
+					// Then for any
+					if(!notNull(numOfTimesProb, startProb, durationProb, probVector, vector)) {
+						numOfTimesProb = nTimesGivenDay.get(ANY);
+						startProb = probStartTime.get(ANY);
+						durationProb = probDuration.get(ANY);
+						probVector = probApplianceUsed.get(ANY);
+						isShiftable = shiftable.get(ANY);
+						isExclusive = config.get(ANY);
+						vector = appliances.get(ANY);
+					}
+				}				
+			}
+		}
+
+		if(vector != null) {
+			int durationMax = Math.min(Constants.MIN_IN_DAY, durationProb.getHistogram().length - 1);
+			for(Appliance app : vector) {
+				//				System.out.println("Appliance:" + app.getName());
+				Double[] applianceConsumption = app.getActiveConsumption();
+				//				NumberFormat nf = new DecimalFormat("0.#");
+				//				for (double c : applianceConsumption) {
+				//					System.out.println(nf.format(c));
+				//				}
+				boolean staticConsumption = app.isStaticConsumption();
+				//				System.out.println("Is static: " + staticConsumption);
+				for (int j = 0; j < act_exp.length; j++)
+					act_exp[j] += aggregatedProbability(durationProb, startProb, applianceConsumption, j, durationMax, staticConsumption);
+				//				for (double c : act_exp) {
+				//					System.out.print(nf.format(c) + " ");
+				//				}
+				//				System.out.println(" ");
+			}
+			for (int j = 0; j < act_exp.length; j++)
+				act_exp[j] = (act_exp[j] * estimateNumberOfTimesFactor(numOfTimesProb) / appliances.size());
+		}
+
+		return act_exp;
+	}
+
+	private double estimateNumberOfTimesFactor (ProbabilityDistribution dailyTimes)
+	{
+		double result = 0;
+		for (int i = 0; i < dailyTimes.getHistogram().length; i++) {
+			result += i * dailyTimes.getHistogram()[i];
+		}
+		return result;
+	}
+
+	private double aggregatedProbability (ProbabilityDistribution duration, ProbabilityDistribution startTime,
+			Double[] consumption, int index,
+			int durationMax,
+			boolean staticConsumption) {
+
+		double result = 0;
+		for (int i = 0; i < durationMax; i++) {
+			if (staticConsumption)
+				result += duration.getProbabilityGreater(i) * startTime.getProbability((Constants.MIN_IN_DAY + index - i) % Constants.MIN_IN_DAY) * consumption[0];
+			else
+				result += duration.getProbabilityGreater(i) * startTime.getProbability((Constants.MIN_IN_DAY + index - i) % Constants.MIN_IN_DAY) * consumption[i % consumption.length];
+		}
+		return result;
+	}
+
+	/**
+	 * Update daily schedule.
+	 *
+	 * @param tick the tick
+	 * @param queue the queue
+	 * @param pricing the pricing policy
+	 * @param baseline the baseline pricing policy
+	 * @param awareness the person's awareness
+	 * @param sensitivity the person's  sensitivity
+	 * @param responseType the response type
+	 * @param orng the random number generator
+	 */
+	public void updateDailySchedule(int tick, PriorityBlockingQueue<Event> queue,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 596-649 into a separate method.